tools/azsdk-cli/docs/specs/3-customizing-client-tsp.spec.md

```diff
@@ -103,44 +164,103 @@ This will live in `eng/common/knowledge/customizing-client-tsp.md` so that all l
+There is an existing [CustomizedCodeUpdateTool](https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/CustomizedCodeUpdateTool.cs) that supports updating customized SDK code after generating an SDK from TypeSpec. Splitting `client.tsp` customization into its own tool (see alternative 1) will put the burden on our users to know whether TypeSpec or SDK code customizations are required to address feedback or breakages. Applying and verifying `client.tsp` customizations should be merged with this existing tool.
 
 Name (CLI): `azsdk tsp customized-update`
-Name (MCP): `azsdk_tsp_update`
+Name (MCP): `azsdk_customized_code_update`
 
-**Current inputs**
+#### Current Inputs
 
 - `--package-path`/`packagePath`: The path to the package (SDK) directory to check. Lives in one of the `azure-sdk-for-*` repos.
 - `--update-commit-sha`/`commitSha`: The SHA of the commit in `azure-rest-api-specs` to use when regenerating the SDK.
-- `--new-gen`(CLI only): Directory for regenerated TypeSpec output (SDK). Optional, defaults to `./tmpgen` within the package path.
 
-**Current behavior**
+#### Current behavior
 
 The tool currently works primarily in an `azure-sdk-for-*` repo.
 
 ```mermaid
 flowchart TD
-    Gen[<b>1. SDK Code Generation</B><br/>• Generate SDK code using updated TypeSpec<br/>• Output to --new-gen directory]
+    Gen[<b>1. SDK Code Generation</B><br/>• Generate SDK code using updated sha<br/>]
 
-    Gen --> Patch[<b>2. Code Customization Updates</b><br/>• AI generates patches from diff + current code<br/>• Apply patches using ClientCustomizationCodePatchTool<br/>• Fallback rules for trivial changes]
+    Gen --> Patch[<b>2. Code Customization Updates</b><br/>• AI generates patches from diff + current code<br/>• Apply patches using ClientCustomizationCodePatchTool<br/>]
 
-    Patch --> Validate[<b>3. Validation and Iteration</b><br/>• Compile SDK code<br/>• If build fails → retry AI patching loop max 2x<br/>• Optional: run tests, surface logical issues]
+    Patch --> Validate[<b>3. Validation and Iteration</b><br/>• Compile SDK code<br/>• If build fails → retry AI patching loop max 2x<br/>]
 
     style Gen fill:#fff9c4
     style Patch fill:#f3e5f5
     style Validate fill:#e8f5e9
 ```
 
-**Proposed Changes**
-TBD
+### Proposed Changes
+
+The existing CustomizedCodeUpdateTool will be enhanced to implement a two-phase customization workflow:
+
+**Enhanced Tool Behavior:**
+
+1. **Phase A – [TypeSpec Customizations](#typespec-customizations):**
+   - Analyze build failures and determine if TypeSpec decorators can resolve issues
```

Comment:
One thing I was thinking with this tool is that it doesn't have to be limited to just build errors. I could imagine passing in API view comments and having the same flow still be applicable.

Is that what you had in mind too? Or do you see this as a tool specializing in build errors?